### PR TITLE
Conv1D and Conv3D supporting dilated conv for CNTK backend

### PR DESCRIPTION
**Pull request by yanboliang**
### Summary
Following #10967, this PR makes ```dilation_rate``` take effect for ```Conv1D``` and ```Conv3D``` for CNTK backend.

### How to test
There is ```test_causal_dilated_conv``` in ```convolutional_test.py```, existing code can't pass when I run against CNTK backend. It's passed after this fix when I run on a node with 2 GPUs. I still keep this test ignored since Travis CI doesn't have GPU node. Another check is to run the following code again TF backend(CPU or GPU) and CNTK backend(GPU), they produce sam result.
```
from keras.layers import Input, Conv1D, Conv2D, Conv3D
from keras.models import Model
from keras.initializers import Constant
import numpy as np

# Test dilated convolution 1D
kernel = np.array([2, 3, 5]).reshape((3, 1, 1))
print(kernel.shape)

inputs = Input(shape=(10, 1))
m = Conv1D(1, kernel_size=3, padding='valid', dilation_rate=2, kernel_initializer=Constant(kernel))(inputs)
model = Model(inputs=inputs, outputs=m)
model.summary()

x = np.arange(10).reshape((1, 10, 1)).astype(np.float32) 
y = model.predict(x)

print(np.squeeze(y))
print(y.shape)

# Test dilated convolution 3D
kernel = np.arange(27).reshape((3, 3, 3, 1, 1))
print(kernel.shape)

inputs = Input(shape=(10, 10, 10, 1))
m = Conv3D(1, kernel_size=3, padding='valid', dilation_rate=(2, 3, 2), kernel_initializer=Constant(kernel))(inputs)
model = Model(inputs=inputs, outputs=m)
model.summary()

x = np.arange(1000).reshape((1, 10, 10, 10, 1)).astype(np.float32) 
y = model.predict(x)

print(np.squeeze(y))
print(y.shape)
```

### PR Overview

- [ ] This PR requires new unit tests [y/n] (make sure tests are included)
- [ ] This PR requires to update the documentation [y/n] (make sure the docs are up-to-date)
- [x] This PR is backwards compatible [y/n]
- [ ] This PR changes the current API [y/n] (all API changes need to be approved by fchollet)

### CODE REVIEW

File: keras/backend/cntk_backend.py [hunk 1]
```diff
@@ -1649,17 +1652,18 @@ def conv3d(x, kernel, strides=(1, 1, 1), padding='valid',
     x = _preprocess_conv3d_input(x, data_format)
     kernel = _preprocess_conv3d_kernel(kernel, data_format)
     padding = _preprocess_border_mode(padding)
-    strides = strides + (strides[0],)
```
Comment (yanboliang) [inline keras/backend/cntk_backend.py]: Here is bug, as even we build a normal convolution(non-dilated) layer, it gets different output compared with TF and TH backend. After the fix, it produces same output.

File: tests/keras/layers/convolutional_test.py [hunk 2]
```diff
@@ -138,8 +140,6 @@ def test_averagepooling_1d():
 
 
 @keras_test
-@pytest.mark.skipif((K.backend() == 'cntk'),
-                    reason="cntk only supports dilated conv on GPU")
```
Comment (yanboliang) [inline tests/keras/layers/convolutional_test.py]: I moved the backend check from here to L179, as we should run other/non-dilated conv test code, they can work on CPU.

File: keras/backend/cntk_backend.py [hunk 3]
```diff
@@ -1494,14 +1494,17 @@ def conv1d(x, kernel, strides=1, padding='valid',
         kernel = C.swapaxes(kernel, 0, 2)
 
     padding = _preprocess_border_mode(padding)
-    strides = [strides]
+
+    if dev.type() == 0 and dilation_rate != 1:
+        raise ValueError('Dilated convolution on CPU is not supported by CNTK backend. ' +
```
Comment (fchollet) [inline keras/backend/cntk_backend.py]: You don't need `+` for string concatenation here.

File: keras/backend/cntk_backend.py [hunk 4]
```diff
@@ -1494,14 +1494,17 @@ def conv1d(x, kernel, strides=1, padding='valid',
...
+                         'Please set dilation_rate with 1.')
```
Comment (fchollet) [inline keras/backend/cntk_backend.py]: `'Please set dilation_date to 1. You passed: %d' % (dilation_rate,)`. And please use backquotes around code keywords.

File: keras/backend/cntk_backend.py [hunk 5]
```diff
@@ -1649,17 +1652,18 @@ def conv3d(x, kernel, strides=(1, 1, 1), padding='valid',
     x = _preprocess_conv3d_input(x, data_format)
     kernel = _preprocess_conv3d_kernel(kernel, data_format)
     padding = _preprocess_border_mode(padding)
-    strides = strides + (strides[0],)
+
+    if dev.type() == 0 and dilation_rate != (1, 1, 1):
+        raise ValueError('Dilated convolution on CPU is not supported by CNTK backend. ' +
+                         'Please set dilation_rate with (1, 1, 1).')
```
Comment (fchollet) [inline keras/backend/cntk_backend.py]: Same comments as above

File: tests/keras/layers/convolutional_test.py [hunk 6]
```diff
@@ -20,7 +20,7 @@
 
 @keras_test
 @pytest.mark.skipif((K.backend() == 'cntk'),
-                    reason="cntk does not support dilated conv")
+                    reason="cntk only support dilated conv on GPU")
```
Comment (fchollet) [inline tests/keras/layers/convolutional_test.py]: Use `'` as quote delimiter, for consistency.

**Pull request merged**